drivers/spi/spim_nrf52.c

```diff
@@ -324,13 +324,13 @@ static int spim_nrf52_init(struct device *dev)
 
 	for (i = 0; i < sizeof(config->psel.ss); i++) {
 		if (config->psel.ss[i] != SS_UNUSED) {
+			spim_nrf52_csn(data->gpio_port, config->psel.ss[i],
```

Comment:
The CS line is never claimed by the SPIM, it is used as a regular GPIO only. See spim_nrf52_csn().
Anyway I updated to use a pull-up as this also fixes the glitch seen on the oscilloscope. But beware that this solution will use slightly more power when toggling CS low as it must drive the additional pull-up.